staging/src/k8s.io/kubectl/pkg/explain/typename.go

```diff
@@ -36,9 +36,9 @@ func (t *typeName) VisitArray(a *proto.Array) {
 	t.Name = fmt.Sprintf("[]%s", s.Name)
 }
 
-// VisitKind just returns "Object".
+// VisitKind returns the path of the kind.
 func (t *typeName) VisitKind(k *proto.Kind) {
-	t.Name = "Object"
+	t.Name = k.GetPath().String()
```

Comment:
`GetPath().String()` is more readable in some contexts, and way less readable in others... (e.g. https://github.com/kubernetes/kubernetes/pull/80387/files#r328156170)